lib/matplotlib/mlab.py

```diff
@@ -520,29 +522,34 @@ def _spectral_helper(x, y=None, NFFT=None, Fs=None, detrend_func=None,
             numFreqs = pad_to//2 + 1
         scaling_factor = 2.
 
+    if not np.iterable(window):
```

Comment:
would `if callbable(window)` make more sense?

Reply:
The logic is the same in apply_window, I don't think handling the behavior change is worth it...